notification_endpoint.go

```diff
@@ -0,0 +1,127 @@
+package influxdb
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+)
+
+var (
+	// ErrInvalidNotificationEndpointType denotes that the provided NotificationEndpoint is not a valid type
+	ErrInvalidNotificationEndpointType = errors.New("unknown notification endpoint type")
+)
+
+// NotificationEndpoint is the configuration describing
+// how to call a 3rd party service. E.g. Slack, Pagerduty
+type NotificationEndpoint interface {
+	Valid() error
+	Type() string
+	json.Marshaler
+	Updator
+	Getter
+	// SecretFields return available secret fields.
+	SecretFields() []SecretField
+	// BackfillSecretKeys fill back fill the secret field key during the unmarshalling
+	// if value of that secret field is not nil.
+	BackfillSecretKeys()
+	// ParseResponse will parse the response of
+	// the http request.
+	ParseResponse(resp *http.Response) error
```

Comment:
I don't think ParseResponse is necessary; my understanding is that Flux is responsible for talking to 3rd party services, not us.